### PR TITLE
cmd: Remove default values for -ethUrl

### LINKED ISSUE #1434: Remove default ethUrl values
**Is your feature request related to a problem? Please describe.**

Currently, the default value for the `-ethUrl` flag is an Infura URL on both Rinkeby and mainnet. This Infura URL will be sunset soon so the node should update its default values.

**Describe the solution you'd like**

Remove the default value for `-ethUrl` flag and require the user to pass in the URL of their preferred ETH RPC provider.

### PR DESCRIPTION
**Pull request by yondonfu**
**What does this pull request do? Explain your changes. (required)**

This PR removes the default values for the `-ethUrl` flag. After this PR, whenever the node connects to an on-chain network, an Ethereum node JSON-RPC URL will need to be explicitly provided using the `-ethUrl` flag.

**Specific updates (required)**

- Remove the default values for the `-ethUrl` flag
- Only run mainnet tests in `test_args.sh` if the `MAINNET_ETH_URL` env variable is set
- Only run Rinkeby tests in `test_args.sh` if the `RINKEBY_ETH_URL` env variable is set
- Use `-ethUrl` in README commands

Values for `MAINNET_ETH_URL` and `RINKEBY_ETH_URL` are set via CircleCI.

**How did you test each of these updates (required)**

Updated tests in `test_args.sh`.

**Does this pull request close any open issues?**

Fixes #1434 

**Checklist:**

- [ ] README and other documentation updated
- [x] Node runs in OSX and devenv
- [x] All tests in `./test.sh` pass

### CODE REVIEW

**Comment by j0sh** [comment]
Curious, will users have a convenient way to discover or acquire an Ethereum JSON-RPC endpoint? Should that be documented somewhere? This is going to require that mainnet users upgrading to the next release make changes to their CLI flags, right?

**Comment by yondonfu** [comment]
Users will still be able to use a hosted node service like Infura or [Alchemy](https://alchemyapi.io/) if they don't want to run their own node they'll just have to sign up for an account themselves.

livepeer/docs#58 updates RTD with a section on connecting either to hosted node service or a self-hosted node. Side note: I was thinking about whether this information should go into this repo or in the RTD and decided to update RTD since some of that information was already there. Would be interested in just rendering go-livepeer specific docs from this repo at some point and having RTD linking to the "go-livepeer guide".

> This is going to require that mainnet users upgrading to the next release make changes to their CLI flags, right?

Yeah. We're removing the default Infura endpoints because we're going to only allow requests from the explorer soon for those endpoints (I believe this is in response to updates to Infura's pricing plan).

**Comment by yondonfu** [comment]
70c2315: Add development docs with notes on running tests and specifying mainnet/Rinkeby Ethereum node JSON-RPC API URLs in order to run certain tests.

**Comment by j0sh** [comment]
> Side note: I was thinking about whether this information should go into this repo or in the RTD and decided to update RTD

Maybe also include a link to the RTD docs for those trying to figure out how to obtain an Ethereum JSON-RPC URL?

I know this has been forewarned a few times on Discord, the forums, etc, but I'd also make sure this change is prominently called out in the release notes and announcement as something that requires user action in order to complete the upgrade. Also doesn't hurt to explain the reason for it, eg Infura pricing, so users feel a little more informed and it seems a little less arbitrary, because registering with a RPC service or setting up a node is going to be a point of friction.

**Review by kyriediculous — APPROVED** [review]
LGTM 👍

**Comment by yondonfu** [comment]
> Maybe also include a link to the RTD docs for those trying to figure out how to obtain an Ethereum JSON-RPC URL?

Added in c808bb4. The link to RTD doesn't actually go to the relevant section, but this will be fixed once livepeer/docs#58 is merged.

> I'd also make sure this change is prominently called out in the release notes and announcement as something that requires user action in order to complete the upgrade. Also doesn't hurt to explain the reason for it, eg Infura pricing, so users feel a little more informed and it seems a little less arbitrary, because registering with a RPC service or setting up a node is going to be a point of friction.

👍 Definitely. Planning on calling this out in the release notes + Discord announcement.

File: doc/ethereum.md
Comment (kyriediculous) [inline doc/ethereum.md]: nit: could we replace "ETH" with "Ethereum" since ETH is more commonly used for the cryptocurrency ?

**Review by kyriediculous — CHANGES_REQUESTED** [review]

File: doc/ethereum.md
Comment (yondonfu) [reply doc/ethereum.md]: Fixed in ef463ef

**Review by j0sh — APPROVED** [review]
LGTM after a fixup

**Review by kyriediculous — APPROVED** [review]

**Comment by kyriediculous** [comment]
Looks good, let's squash the fixup commit 👍

**Pull request merged**
